x/gamm/keeper/grpc_query.go

```diff
@@ -288,6 +288,29 @@ func (q Querier) CalcExitPoolCoinsFromShares(ctx context.Context, req *types.Que
 	return &types.QueryCalcExitPoolCoinsFromSharesResponse{TokensOut: exitCoins}, nil
 }
 
+//  CalcJoinPoolNoSwapShares returns the amount of shares you'd get if joined a pool without a swap and tokens which need to be provided
+func (q Querier) CalcJoinPoolNoSwapShares(ctx context.Context, req *types.QueryCalcJoinPoolNoSwapSharesRequest) (*types.QueryCalcJoinPoolNoSwapSharesResponse, error) {
```

Comment:
Can we please add this and your previous query to this test: https://github.com/osmosis-labs/osmosis/blob/0423e96433742f09ca2f71e286be2375e5fd90a4/x/gamm/client/cli/query_test.go#L26

It ensures that queries don't alter state. This would be very useful for a stargate query

Reply:
Got it, I'll change this ASAP